errors/errors.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package errors
+
+import "fmt"
+
+// Error specifies an API that must be fullfiled by error type
+type Error interface {
+
+	// Error implements the error interface.
+	Error() string
+
+	// Msg returns error message
+	Msg() string
+
+	// Err returns wrapped error
+	Err() *customError
+}
+
+var _ Error = (*customError)(nil)
+
+// customError struct represents a Mainflux error
+type customError struct {
+	msg string
+	err *customError
```

Comment:
Maybe we should use `Error` here instead of `*customError`. Later you can chain some other errors that implement this interface here. I think that using interface is more flexible.